tests/helpers_tests/test_audience.py

```diff
@@ -171,12 +181,92 @@ def test_is_user_in_experiment__evaluates_audience_conditions_leaf_node(self):
     experiment.audienceConditions = '3468206645'
 
     with mock.patch('optimizely.helpers.condition.CustomAttributeConditionEvaluator') as custom_attr_eval:
-      audience.is_user_in_experiment(project_config, experiment, {})
+      audience.is_user_in_experiment(project_config, experiment, {}, self.mock_client_logger)
 
     audience_3468206645 = project_config.get_audience('3468206645')
 
     custom_attr_eval.assert_has_calls([
-        mock.call(audience_3468206645.conditionList, {}),
+        mock.call(audience_3468206645.conditionList, {}, self.mock_client_logger),
         mock.call().evaluate(0),
         mock.call().evaluate(1),
     ], any_order=True)
+
+
+class AudienceLoggingTest(base.BaseTest):
+
+  def setUp(self):
+    base.BaseTest.setUp(self)
+    self.mock_client_logger = mock.MagicMock()
+
+  def test_is_user_in_experiment__with_no_audience(self):
+    experiment = self.project_config.get_experiment_from_key('test_experiment')
+    experiment.audienceIds = []
+    experiment.audienceConditions = []
+
+    with mock.patch('optimizely.logger.reset_logger', return_value=self.mock_client_logger):
```

Comment:
Nit: does `optimizely.logger.reset_logger` actually need to return `self.mock_client_logger` given that we are passing `self.mock_client_logger` directly to the function we're testing, on the line below?

Same question elsewhere in the new tests.